cluster/src/main/java/com/hartwig/pipeline/execution/vm/BucketCompletionWatcher.java

```diff
@@ -0,0 +1,49 @@
+package com.hartwig.pipeline.execution.vm;
+
+import java.time.Duration;
+import java.util.List;
+
+import com.google.cloud.storage.Blob;
+import com.hartwig.pipeline.storage.RuntimeBucket;
+
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+
+public class BucketCompletionWatcher {
+    enum State {
+        SUCCESS,
+        FAILURE,
+        STILL_WAITING
+    }
+
+    State waitForCompletion(RuntimeBucket bucket, VirtualMachineJobDefinition jobDefinition) {
+        return Failsafe.with(new RetryPolicy<>().withMaxRetries(-1).withDelay(Duration.ofSeconds(5)).handleResult(null)).get(() -> {
+            State currentState = currentState(bucket, jobDefinition);
+            if (currentState.equals(State.STILL_WAITING)) {
+                return null;
+            } else {
+                return currentState;
+            }
+        });
+    }
+
+    State currentState(RuntimeBucket bucket, VirtualMachineJobDefinition jobDefinition) {
```

Comment:
Maybe this could be split up even further? No code relationship between these two exposed methods